x-pack/plugin/repository-encrypted/src/main/java/org/elasticsearch/repositories/encrypted/DecryptionPacketsInputStream.java

```diff
@@ -124,19 +123,19 @@ public void reset() throws IOException {
         throw new IOException("Mark/reset not supported");
     }
 
+    @Override
+    public void close() throws IOException {
+        IOUtils.close(super::close, source);
+    }
+
     private int decrypt(PrefixInputStream packetInputStream) throws IOException {
         // read only the IV prefix into the packet buffer
         int ivLength = packetInputStream.readNBytes(packetBuffer, 0, GCM_IV_LENGTH_IN_BYTES);
         if (ivLength != GCM_IV_LENGTH_IN_BYTES) {
             throw new IOException("Packet heading IV error. Unexpected length [" + ivLength + "].");
         }
-        // extract the nonce and the counter from the packet IV
-        ByteBuffer ivBuffer = ByteBuffer.wrap(packetBuffer, 0, GCM_IV_LENGTH_IN_BYTES).order(ByteOrder.LITTLE_ENDIAN);
-        int packetIvNonce = ivBuffer.getInt(0);
-        long packetIvCounter = ivBuffer.getLong(Integer.BYTES);
-        if (packetIvNonce != nonce) {
-            throw new IOException("Packet nonce mismatch. Expecting [" + nonce + "], but got [" + packetIvNonce + "].");
-        }
```

Comment:
Removing the explicit nonce check was necessary in order to support using the same DEK for multiple encrypted blobs.

Previously, every blob had its own associated metadata blob, where we stored the nonce, which was explicitly verified here, during decryption. But now there is no place where we can associate metadata for every blob (and this is by design, because associating something to every blob incurs at least one API call and it's difficult to update).

There is no security problem if we're not explicitly checking the nonce, because the nonce is part of every packet's IV and is hence validated when the packet's authn tag is validated, implicitly by the GCM algorithm.

Reply:
That makes sense.
Is there still a reason why we store it in the stream then?